### PR TITLE
Create new offer component

### PR DESCRIPTION
**Pull request by Angham116**
### Related:
related to #27 

### Notes: 
   how to test this PR:
     - ```git fetch origin create-offer-component```
     - ```git checkout create-offer-component```
     - ```cd client/ ```
     - ```npm i```

### Checklist Criteria
- [x] **Branch is up to date with master**
- [x] **The project runs**
- - [x] You can download and install locally, and run the project locally
- - [x] All existing and new features run without bugs
- - [x] The build script runs without error
- [x] **Code is consistent**
- - [x] Linter is used, no error messages
- - [x] Code is consistent with agreed team style
- - [x] Naming of files is consistent with project (e.g. all files named with camelCase)
- [x] **No console logs**
- - [x] Console.log is only used for crucial system messages (e.g. which port server is on).
- - [x] Any console.logs used for debugging have been removed.

### CODE REVIEW

File: client/src/Components/CreateOffer/index.jsx [hunk 1]
```diff
@@ -1,11 +1,186 @@
...
+          this.setState({ errMsg: { ...errors } });
+        }
+      });
+  };
+
+  handleTitle = title => this.setState({ title });
+
+  handlePosition = position => this.setState({ position });
+
+  handleDescription = description => this.setState({ description });
```
Comment (ashatat) [inline client/src/Components/CreateOffer/index.jsx]: group handleTitle, handlePosition, handleDescription in one function, something like:
```js
handleInput = event => {
  const { name, value } = event.target;
  this.setState({ [name]: value });
};
```
also you'd need to change it on the component tag soemthing like this
```jsx
<Form.Control
  type="text"
  name="title"
  placeholder="eg:  Small Shop seeks workers "
  onChange={this.handleInput}
 />
```

File: client/src/Components/CreateOffer/index.jsx [hunk 2]
```diff
@@ -1,11 +1,186 @@
...
+                      className="newoffer-btn--apply"
+                      onClick={this.handleSubmit}
+                    >
+                      Create
+                    </Button>
+                  </Col>
+                  <Col sm={12} md={12} lg={1} className="newoffer--btn">
+                    <Button
+                      className="newoffer-btn--cancel"
+                      onClick={() => history.push('/home')}
```
Comment (ashatat) [inline client/src/Components/CreateOffer/index.jsx]: maybe it'd be better if we return the user back to the previous page he was on?

File: client/src/Components/CreateOffer/style.css [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+.newoffer__content {
+  padding-left: 0.7rem;
+}
+
+.newoffer__label{
+  color: rgba(51, 51, 51, 0.8);
+}
+
+.newoffer--description{
```
Comment (ashatat) [inline client/src/Components/CreateOffer/style.css]: :sob:

File: client/src/Components/CreateOffer/style.css [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  width: 70% !important;
+  margin-bottom: 1rem;
+}
+
+.newoffer__btns {
+  margin-bottom: 2rem;
+  margin-top: 1.8rem;
+}
+
+.newoffer-btn--cancel,
```
Comment (ashatat) [inline client/src/Components/CreateOffer/style.css]:
```suggestion
.newoffer__btn--cancel,
```

File: client/src/Components/CreateOffer/index.jsx [hunk 5]
```diff
@@ -1,11 +1,186 @@
...
+                    </div>
+                  )}
+                  <Form.Label className="newoffer__label">
+                    {' '}
+                    Position:{' '}
+                  </Form.Label>
+                  <Form.Control
+                    type="text"
+                    placeholder="eg:   Marketing Manager"
+                    onChange={({ target }) => this.handlePosition(target.value)}
```
Comment (ashatat) [inline client/src/Components/CreateOffer/index.jsx]: this way you are using an arrow function here also another function `this.handlePosition` which is not necessary

File: client/src/Components/CreateOffer/index.jsx [hunk 6]
```diff
@@ -1,11 +1,186 @@
...
+    const { title, position, description, offerType } = this.state;
+
+    newOfferValidation
+      .validate(
+        { title, position, description, offerType },
+        { abortEarly: false }
+      )
+      .then(() => {
+        this.setState({ errMsg: '' });
+        console.log(this.state);
```
Comment (ashatat) [inline client/src/Components/CreateOffer/index.jsx]: :sob:  remove console.log

File: client/src/Components/CreateOffer/style.css [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  margin-bottom: 1rem;
+}
+
+.newoffer__btns {
+  margin-bottom: 2rem;
+  margin-top: 1.8rem;
+}
+
+.newoffer__btn--cancel,
+.newoffer-btn--apply {
```
Comment (ashatat) [inline client/src/Components/CreateOffer/style.css]: :man_facepalming: :woman_facepalming:

**Comment by AymanAlqoqa** [comment]
image
please, add some margin between buttons
thanks

**Comment by Angham116** [comment]
> image
> please, add some margin between buttons
> thanks

**Comment by Angham116** [comment]
> image
> please, add some margin between buttons
> thanks

@AymanAlqoqa  i think we said will make the design later :)

File: client/src/Components/CreateOffer/index.jsx [hunk 8]
```diff
@@ -1,11 +1,179 @@
...
+import './style.css';
+
+class CreateOffer extends Component {
+  state = {
+    title: '',
+    position: '',
+    description: '',
+    skills: null, // skills id's
+    offerType: null, // offer_types id's
+    // member_id: '',
```
Comment (Fatmasiam) [inline client/src/Components/CreateOffer/index.jsx]: command code

File: client/src/Components/CreateOffer/index.jsx [hunk 9]
```diff
@@ -1,11 +1,179 @@
...
+    errMsg: '',
+  };
+
+  componentDidMount() {
+    // const member_id = get member_id from local storage
```
Comment (Fatmasiam) [inline client/src/Components/CreateOffer/index.jsx]: command code

File: client/src/Components/CreateOffer/index.jsx [hunk 10]
```diff
@@ -1,11 +1,179 @@
...
+    // this.setState({ member_id })
+  }
+
+  handleSubmit = () => {
+    // eslint-disable-next-line react/prop-types
+    // const { history } = this.props;
```
Comment (Fatmasiam) [inline client/src/Components/CreateOffer/index.jsx]: and here you have command code

File: client/src/Components/CreateOffer/index.jsx [hunk 11]
```diff
@@ -1,11 +1,179 @@
...
+    newOfferValidation
+      .validate(
+        { title, position, description, offerType },
+        { abortEarly: false }
+      )
+      .then(() => {
+        this.setState({ errMsg: '' });
+        // send request to the backend with body
+        // axios
+        // .post('/api/v1/offers', { title, position, description, skills, offerType })
```
Comment (Fatmasiam) [inline client/src/Components/CreateOffer/index.jsx]: command code. 
just write a short explanation of what will happen

File: client/src/Components/CreateOffer/index.jsx [hunk 12]
```diff
@@ -1,11 +1,179 @@
...
+  };
+
+  handleSkills = skills => this.setState({ skills });
+
+  handleOfferTypes = offerType => this.setState({ offerType });
+
+  handleInput = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
```
Comment (Fatmasiam) [inline client/src/Components/CreateOffer/index.jsx]: just suggestion 
```javascript 
handleInput =({target : {name, value} }) => this.setState({ [name] : value}) ;
```

File: client/src/Components/CreateOffer/index.jsx (hunk 8 above)
Comment (ashatat) [reply client/src/Components/CreateOffer/index.jsx]: @Fatmasiam  command?

File: client/src/Components/CreateOffer/index.jsx (hunk 11 above)
Comment (ashatat) [reply client/src/Components/CreateOffer/index.jsx]: oh you mean commented code!! :thinking: 

I think it's ok the commented code here explain itself

Comment (Fatmasiam) [reply client/src/Components/CreateOffer/index.jsx]: @ashatat  yeah :sweat_smile:

Comment (Angham116) [reply client/src/Components/CreateOffer/index.jsx]: @Fatmasiam I need all of these comments when joining front with back-end

**Pull request merged**